src/CLI.hs

```diff
@@ -341,6 +377,62 @@ runCLI args = handle handler $ do
       Nothing -> do
         logDebug "Reading from stdin"
         getContents' `catch` (\(e :: SomeException) -> throwIO (CouldNotReadFromStdin (show e)))
+    performMatch' :: String -> Maybe Condition -> Program -> PrintMode -> IO String
+    performMatch' patternStr maybeCond prog@(Program rootExpr) mode = do
```

Comment:
@yarkivaev the functionality of matching program with rule is already implemented in `Rule` module, the function `matchProgramWithRule` is what you need